From f700d2c292f18946b68b9311e9e7852720490b35 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Tue, 12 Jan 2016 14:49:00 +0200
Subject: [PATCH 0666/1240] fix: iob: Fix end address printout in dump_iob

	First IOB window always points to configuration space, and it's size is
	hardcoded to 16MB.

Change-Id: Ie54524b0ed6adafab3d18584f1a5cdd94a568d5b
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26733
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/cpu/armv8/armada8k/mvebu_iob.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada8k/mvebu_iob.c b/arch/arm/cpu/armv8/armada8k/mvebu_iob.c
index 8372305..b510170 100644
--- a/arch/arm/cpu/armv8/armada8k/mvebu_iob.c
+++ b/arch/arm/cpu/armv8/armada8k/mvebu_iob.c
@@ -139,9 +139,15 @@ void dump_iob(void)
 		if (win_cr & WIN_ENABLE_BIT) {
 			target_id = (win_cr >> IOB_TARGET_ID_OFFSET) & IOB_TARGET_ID_MASK;
 			alr = readl(IOB_WIN_ALR_OFFSET(win_id));
-			ahr = readl(IOB_WIN_AHR_OFFSET(win_id));
 			start = ((u64)alr << ADDRESS_SHIFT);
-			end = (((u64)ahr + 0x10) << ADDRESS_SHIFT);
+			if (win_id != 0) {
+				ahr = readl(IOB_WIN_AHR_OFFSET(win_id));
+				end = (((u64)ahr + 0x10) << ADDRESS_SHIFT);
+			} else {
+				/* Window #0 size is hardcoded to 16MB, as it's
+				** reserved for CP configuration space. */
+				end = start + (16 << 20);
+			}
 			printf("iob   %02d %s   0x%016llx 0x%016llx\n"
 					, win_id, iob_target_name[target_id], start, end);
 		}
-- 
1.9.1

